scipy/signal/_signaltools.py

```diff
@@ -4625,3 +4625,142 @@ def decimate(x, q, n=None, ftype='iir', axis=-1, zero_phase=True):
 
     return y[tuple(sl)]
+
+def envelope(x,N = None,method = 'analytic'):
+    """
+    Calculate the upper and lower envelopes of a 1-D signal.
+
+    By default, the envelopes of `x` are calculated by the magnitude of its
+    analytical signal.
+
+    Parameters
+    ----------
+    x : array_like
+        The signal for which the envelopes are going to be calculated, as a
+        1-dimensional array.
+    N : int, optional
+        If `method` = `'analytic'`, `N` is the number of Fourier components. If
+        `method` = `'rms'`, `N` is the sliding window length. If `method` =
+        `'peak'`, `N` is the required minimal horizontal distance between
+        neighbouring peaks. For the `'rms'` and `'peak'` methods `N` must be
+        specified and must be positive. Default: ``x.shape[0]``
+    method : str {`'analytic'`, `'rms'`, `'peak'`}, optional
+        A string indicating which method to use to calculate the envelopes.
+
+        ``analytic``
+           The envelopes of x are calculated by the magnitude of its
+           analytical signal.
+        ``rms``
+           Calculates the upper and lower root-mean-square envelopes of `x`. The
+           mean is determined by a sliding window of `N` samples.
+        ``peak``
+           Calculates the upper and lower peak envelopes of `x`. The envelopes
+           are determined by employing spline interpolation between adjacent
+           local maxima that are at least `N` samples apart.
+
+    Returns
+    -------
+    upper : ndarray
+        The upper envelope of the signal.
+    lower : ndarray
+        The lower envelope of the signal.
+
+    See Also
+    --------
+    hilbert : Compute the analytic signal, using the Hilbert transform.
+    find_peaks : Find peaks inside a signal based on peak properties.
+
+    Examples
+    --------
+
+    >>> import numpy as np
+    >>> from scipy.signal import envelope
+    >>> import matplotlib.pyplot as plt
+
+    Create a synthetic signal and calculate its analytical envelopes.
+
+    >>> t = np.arange(0, 2, 1/2000)
+    >>> signal = (1+0.5*np.cos(2*np.pi*1*t))*np.cos(2*np.pi*10*t)+5
+    >>> upper,lower = envelope(signal)
+    >>> plt.plot(t,signal)
+    >>> plt.plot(t,upper)
+    >>> plt.plot(t,lower)
+    >>> plt.show()
+
+    Create a synthetic signal and calculate its root-mean-square envelopes.
+
+    >>> t = np.arange(0, 2, 1/2000)
+    >>> signal = (1+0.5*np.cos(2*np.pi*1*t))*np.random.normal(size = t.shape[0])+5
+    >>> upper,lower = envelope(signal,N = 150,method = 'rms')
+    >>> plt.plot(t,signal)
+    >>> plt.plot(t,upper)
+    >>> plt.plot(t,lower)
+    >>> plt.show()
+
+    .. versionadded:: 1.12.0
+    """
+    from ._peak_finding import find_peaks
+
+    #Assert that x is a 1D array
+    x = np.asarray(x)
+    assert isinstance(x, np.ndarray) and x.ndim == 1\
+            , 'x must be a 1-dimensional array'
+
+    if N or N==0: #N is not None.
+        assert isinstance(N,int) and N>0,\
```

Comment:
No asserts, assume N is int by passing it through `N = int(N)`. If fails it's a user problem. Put it in the documentation that strictly integer valued inputs are accepted.